.github/workflows/reusable-cpp-coverage.yml

```diff
@@ -22,6 +22,14 @@ on:
         description: "The version of Z3 to set up"
         default: "4.13.4"
         type: string
+      setup-mlir:
+        description: "Whether to set up MLIR"
+        default: false
+        type: boolean
+      setup-mlir-tag:
+        description: "The setup-mlir release tag"
+        default: 2025.11.25
+        type: string
```

Comment:
This still feels somewhat wrong to me. I would really expect to set an LLVM version here.
Most conveniently, I would set something like `21.1.6` and the action would check and see if it finds that version. For that, it would crawl the releases from newest to oldest and see whether it finds a matching version. I would guess that setup-uv does something similar.
Does that sound reasonable?